### PR TITLE
fix: upgrade node v14 to v18

### LINKED ISSUE #1364: Upgrade from node14 to 18
# Description

We ran into error like this: 

```
[semantic-release]: node version >=18 is required. Found v14.21.2.
```
Please help us upgrade the node version to 18, and test if anything is good.

To reproduce

[comment]: # 'To reproduce'

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by keyy123**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

Upgrade node to version 18 in all .yaml files in attempt to fix semantic release error about using node >= 18

Fixes #1364

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
- [X] Jest unit tests

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by dadiorchen** [comment]
@keyy123 after merging your pr, I think we solve the previous problem, but there are something new: 
https://github.com/Greenstand/treetracker-web-map-client/actions/runs/4035507878/jobs/6937465226

What do you think?

**Comment by keyy123** [comment]
The issue is the `npm ci` task within the `Build Front End Project` job is reading the engine prop within package*.json. I don't know which one but it looks like `node: ^16` .  

The `npm ci` job is equivalent to setting `engine-strict=true` in an `.npmrc` file meaning that if the version does not exactly match what version is in engine object instead of returning warning from npm it will return errors showing the difference between what is required and what you tried to install in the project and ending  process with exit code 1. 

TLDR: We just need to make sure the node version in engine is carefully specified within `package.json` even if its a range of possible node versions.

**Comment by keyy123** [comment]
I can handle the issue immediately, if you tell me which versions of node that you want to application to use.

**Comment by dadiorchen** [comment]
I think we can go node 18 if it is possible?

**Comment by keyy123** [comment]
I just made a PR for the follow up issue. How should I handle it @dadiorchen ?

**Comment by dadiorchen** [comment]
How about you try to do it by yourself? I have given you the write permission for this repo.

**Comment by dadiorchen** [comment]
So you can merge and check the action result

**Comment by keyy123** [comment]
Good news @dadiorchen , It passed the check for npm ci BUT it failed due to hitting its rate limit. I will retry the job after a few minutes based on this issue ticket on semantic_release repo. Unfortunately, It may also indicate that Semantic Release is having issues b/c Github may have updated its policies. They propose a potential fix to the error in February.

TLDR: GitHub may have changed its rules and its causing Semantic Release to error with the rate limit error. It seems random to some devs and persistent to others. I will push an empty commit within 10 minutes to confirm info from semantic release ticket.

**Comment by dadiorchen** [comment]
But I saw it passed, right? https://github.com/Greenstand/treetracker-web-map-client/actions/runs/4040509103 after some trying I guess

**Comment by dadiorchen** [comment]
Good job and observation!
